### PR TITLE
feat: add lens to zoom in on sky, showing molecules

### LINKED ISSUE #28: add lens which zooms in on molecules
two options depending on what's feasible:

- a static lens fixed on a given position in the canvas, with some molecules inside the lens area (indicating that this is a zoomed-in view, in contrast to the rest of the canvas)
- a lens that's draggable over the canvas which zooms in on the area it is over (showing the molecules therein)

### PR DESCRIPTION
**Pull request by hagoptaminian**
closes #28

### CODE REVIEW

**Comment by hagoptaminian** [comment]
two notes:

(1) There are some gymnastics needed to get the cursor behaving correctly. This is why you'll see `onMouseEnter` and `onMouseLeave` events in different parts of the canvas. I can elaborate on a zoom call if needed, but basically, the challenge is: `onMouseEnter` or `onMouseMove` require the mouse to move to update the cursor. But, if you have just clicked zoom in, you want the cursor to *automatically* change to zoom-out (not having to move the mouse) - otherwise it's quite ugly if you've just clicked zoom in and it's still a zoom-in cursor. So that motivates this seemingly weird approach. 

The one edge case where there is still misbehavior: Click zoom in --> Scroll to somewhere near the top (or the bottom) of the canvas --> click zoom out --> The cursor will be a zoom-in cursor even though it's on top of the atmosphere/ground/sea

If you find a better solution, let me know!

(2) Minor: Quite rarely the molecules within a row in the zoomed in view can overlap (if the random coordinates happen to be quite close). Obviously this is fixable but since it's quite rare I thought we can leave it for another issue if needed.

**Comment by pyphilia** [comment]
Thanks  @hagoptaminian for this PR! 😄  I'll take a look soon, is that okay for me to take it over or do you wish my review to refine it yourself?

**Comment by pyphilia** [comment]
@hagoptaminian  I think I won't work on it until next week, so if you have time and you like to go further with it, go ahead^^ we can catch up with a call if necessary!

File: src/components/lab/canvas/Sea.js [hunk 1]
```diff
@@ -29,7 +29,12 @@ const Sea = ({ stageHeight, stageWidth }) => {
   const seaBeginsY = atmosphereAndSkyHeight;
 
   return (
-    <Group>
+    <Group
+      onMouseEnter={(event) => {
```
Comment (pyphilia) [inline src/components/lab/canvas/Sea.js:33]: Add comments about what you talk about in the PR's conversation.

File: src/components/lab/canvas/Sky.js [hunk 2]
```diff
@@ -22,7 +21,12 @@ const Sky = ({ stageHeight, stageWidth }) => {
   const skyBeginsY = atmosphereHeight;
 
   return (
-    <Group>
+    <Group
+      onMouseEnter={(event) => {
+        const container = event.target.getStage().container();
+        container.style.cursor = 'zoom-in';
```
Comment (pyphilia) [inline src/components/lab/canvas/Sky.js:27]: Factor out the cursor type.

File: src/components/lab/canvas/sky/greenhouse-gases/molecules/CarbonDioxide.js [hunk 3]
```diff
@@ -25,7 +25,16 @@ const CarbonDioxide = ({ moleculeCenter, skyHeight }) => {
...
   return (
-    <Group>
+    <Group
+      onMouseEnter={(event) => {
+        const container = event.target.getStage().container();
+        container.style.cursor = 'default';
+      }}
+      onMouseLeave={(event) => {
+        const container = event.target.getStage().container();
+        container.style.cursor = 'zoom-out';
```
Comment (pyphilia) [inline src/components/lab/canvas/sky/greenhouse-gases/molecules/CarbonDioxide.js:35]: Factor out.

**Review by pyphilia — APPROVED** [review]
Thank you @hagoptaminian 💯 
I'm wondering about the `onMouseEnter` `onMouseLeave`, if it wouldn't be clearer to pass down the functions from `CanvasLayout`. It would gather all the functions (which are pretty much the same) in the same place. I think it would be clearer that it implements the cursor behavior. Let me know what you think.

**Comment by hagoptaminian** [comment]
@pyphilia good call on factoring out the onMouse handlers to avoid repetition 👍 Did this and the other two suggestions.

**Pull request merged**